### PR TITLE
Install clusterctl binary with curl

### PR DESCRIPTION
**Pull request by furkatgofurov7**
This patch moves clusterctl to 01_* and installs it using binary

### CODE REVIEW

File: 01_prepare_host.sh [hunk 1]
```diff
@@ -79,6 +79,12 @@ if ! command -v kustomize 2>/dev/null ; then
     rm "kustomize_${KUSTOMIZE_VERSION}_linux_amd64.tar.gz"
 fi
 
+if ! [ -x "$(command -v clusterctl)" ]; then
+  curl -L -O "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CLUSTERCTL_VERSION}/clusterctl-linux-amd64 -o clusterctl"
```
Comment (fmuyassarov) [inline 01_prepare_host.sh:83]:
```suggestion
  curl -L -O "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CLUSTERCTL_VERSION}/clusterctl-linux-amd64" -o clusterctl
```

Comment (furkatgofurov7) [reply 01_prepare_host.sh]: Fixed, thanks!

File: 01_prepare_host.sh [hunk 2]
```diff
@@ -79,6 +79,12 @@ if ! command -v kustomize 2>/dev/null ; then
     rm "kustomize_${KUSTOMIZE_VERSION}_linux_amd64.tar.gz"
 fi
 
+if ! [ -x "$(command -v clusterctl)" ]; then
+  curl -L -O "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CLUSTERCTL_VERSION}/clusterctl-linux-amd64" -o clusterctl
```
Comment (fmuyassarov) [inline 01_prepare_host.sh:83]:
```suggestion
  curl -L -O https://github.com/kubernetes-sigs/cluster-api/releases/download/"${CLUSTERCTL_VERSION}"/clusterctl-linux-amd64 -o clusterctl
```
Maybe this will work?

File: lib/common.sh [hunk 3]
```diff
@@ -178,9 +178,13 @@ else
   echo "Management cluster forced to be minikube when container runtime is not docker"
   export EPHEMERAL_CLUSTER="minikube"
 fi
+
 #Kustomize version
 export KUSTOMIZE_VERSION=${KUSTOMIZE_VERSION:-"v3.6.1"}
 
+#clusterctl version
+export CLUSTERCTL_VERSION=${CLUSTERCTL_VERSION:-"v0.3.8"}
```
Comment (maelk) [inline lib/common.sh:186]: You should probably use the CAPIRELEASE variable to get the latest CAPI release (you need to import lib/releases.sh)

File: 01_prepare_host.sh (hunk 2 above)
Comment (maelk) [reply 01_prepare_host.sh:83]: it might be slightly more readable with the quotes outside ?

File: lib/releases.sh [hunk 4]
```diff
@@ -7,6 +7,9 @@ function get_latest_release() {
   else
     release="$(curl -H "Authorization: token ${GITHUB_TOKEN}" -sL "${1}")" || ( set -x && exit 1 )
   fi
+  if [ "${EPHEMERAL_CLUSTER}" == "minikube" ]; then
+    sudo yum install jq -y
+  fi
```
Comment (maelk) [inline lib/releases.sh:12]: why is this here ? This file is only fetching the releases. if jq is not here, it means that it is sourced at the wrong place in 01_ . In addition, minikube != centos, you can run on Ubuntu with minikube

File: 01_prepare_host.sh [hunk 5]
```diff
@@ -79,6 +81,12 @@ if ! command -v kustomize 2>/dev/null ; then
     rm "kustomize_${KUSTOMIZE_VERSION}_linux_amd64.tar.gz"
 fi
 
+if ! [ -x "$(command -v clusterctl)" ]; then
+  curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/"${CAPIRELEASE}"/clusterctl-linux-amd64 -o clusterctl
+  chmod +x ./clusterctl
+  sudo mv ./clusterctl /usr/local/bin/clusterctl
+fi
```
Comment (maelk) [inline 01_prepare_host.sh:88]: what if CAPIRELEASE changes in between two runs ? you should probably check the version and if it is not in sync with the release version, upgrade.

Comment (fmuyassarov) [reply 01_prepare_host.sh]: Fixed, thanks.

File: lib/releases.sh (hunk 4 above)
Comment (fmuyassarov) [reply lib/releases.sh]: Fixed.

**Comment by fmuyassarov** [comment]
As @furkatgofurov7 on leave right now, I will try to finalize the PR.
